src/index.ts

```diff
@@ -40,7 +40,7 @@ const access = (absWithExt: string): Promise<string> => {
  * @return {Promise<string>} Resolves the absolute file path just checked, or undefined.
  */
 const isExecutable = async (abspath: string): Promise<string> => {
-    const checks = getApplicableExtensions().map(ext => access(abspath + ext));
+    const checks = getApplicableExtensions().map(ext => access(abspath.endsWith(ext) ? abspath : abspath + ext));
```

Comment:
This will not work as `endsWith` is case-sensitive. Note that `getApplicateExensions` may return `EXE` instead of `exe`.

```suggestion
    const checks = getApplicableExtensions().map(ext => access(abspath.toLocaleLowerCase().endsWith(ext.toLocaleLowerCase()) ? abspath : abspath + ext));
```

Reply:
Cool. Good to know. Thank you very much.